library/manageiq_provider.py

```diff
@@ -531,16 +551,18 @@ def main():
             provider_region=dict(required=False, type='str'),
             access_key_id=dict(required=False, type='str', no_log=True),
             secret_access_key=dict(required=False, type='str', no_log=True),
-            metrics=dict(required=False, type='bool', default=False),
-            hawkular_hostname=dict(required=False),
-            hawkular_port=dict(required=False),
+            monitoring=dict(required=False, default=None,
+                       choices=['hawkular', 'prometheus', None]),
+            monitoring_hostname=dict(required=False),
+            monitoring_port=dict(required=False),
             initiate_refresh=dict(required=False, type='bool', default=True),
             validate_provider_auth=dict(required=False, type='bool', default=True)
         ),
         required_if=[
             ('provider_type', 'openshift-origin', ['provider_api_hostname', 'provider_api_port', 'provider_api_auth_token']),
             ('provider_type', 'openshift-enterprise', ['provider_api_hostname', 'provider_api_port', 'provider_api_auth_token']),
-            ('metrics', True, ['hawkular_hostname', 'hawkular_port']),
+            ('monitoring', 'hawkular', ['monitoring_hostname', 'monitoring_port']),
+            ('monitoring', 'prometheus', ['monitoring_hostname', 'monitoring_port']),
```

Comment:
I am not so happy with writing this validation twice - once for prometheus and once for hawkular, but I couldn't find a way in `required_if` to state "not None". Suggestions are welcome...